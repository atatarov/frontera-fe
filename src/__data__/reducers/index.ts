import { combineReducers } from 'redux';
import suggestions from './suggestions';
import trending from './trending'

export default combineReducers({
    suggestions,
    trending
});
