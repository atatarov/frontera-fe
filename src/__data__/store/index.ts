import { configureStore } from '@reduxjs/toolkit';

import { createAPI } from 'services/api';

import reducer from '../reducers';

const middlewareOptions: any = {
  thunk: true,
  immutableCheck: false,
  serializableCheck: false
};

export const api = createAPI();

export const store = configureStore({
  reducer,
  devTools: {
    name: 'Frontera'
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      ...middlewareOptions,
      thunk: {
        extraArgument: api
      }
    })
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
