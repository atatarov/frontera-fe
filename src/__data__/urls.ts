import { getNavigations } from '@ijl/cli';

const navigations = getNavigations('frontera');

export const baseUrl = ['frontera'];

export const URLs = {
    login: {
        url: navigations['link.frontera.login']
    },
    lang: {
        url: navigations['link.frontera.lang']
    },
    themes: {
        url: navigations['link.frontera.themes']
    },
    personalizing: {
        url: navigations['link.frontera.personalizing']
    },
    suggestions: {
        url: navigations['link.frontera.suggestions']
    },
    home: {
        url: navigations['link.frontera.home']
    },
    registration: {
        url: navigations['link.frontera.registration']
    },
    resetpassword: {
        url: navigations['link.frontera.resetpassword']
    },
}