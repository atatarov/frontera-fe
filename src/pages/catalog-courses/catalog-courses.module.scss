.container {
  width: 1200px;
  margin: 0 auto;
  padding: var(--offset-small) 0;
}

@media (min-width: 900px) and (max-width: 1279.98px) {
  .container {
    width: 776px;
  }
}

@media (min-width: 768px) and (max-width: 899.98px) {
  .container {
    width: 648px;
  }
}

@media (max-width: 767.98px) {
  .container {
    width: 320px;
  }
}