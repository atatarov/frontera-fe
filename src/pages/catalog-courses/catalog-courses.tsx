import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';

import Header from '../../components/header';
import { PageHeader } from '../../components/page-header';
import Loader from "../../components/loader";

import { CourseCardsLarge } from '../../components/course-cards/course-cards';
import getSuggestion from '../../__data__/action/suggestions';

import style from './style.css';

export const CatalogCourses = () => {
    const { t } = useTranslation();
    const dispatch = useDispatch();
    const { isLoading } = useSelector((state: any) => ({ isLoading: state.suggestions.loading }));
    const { courseList } = useSelector((state: any) => ({ courseList: state.suggestions.courseList }));
    const { errors } = useSelector((state: any) => ({ errors: state.suggestions.errors }));

    useEffect(() => {
        if (!isLoading && courseList === null) {
            dispatch(getSuggestion());
        }
    }, []);


    return (
        <React.Fragment>
            <Header />
            <main className={style.container}>
                <PageHeader label={t('catalog.courses.title')} />
                <section>
                    {isLoading ? <Loader />
                        : courseList && <CourseCardsLarge showNumber={false} cards={courseList} />}
                    {errors && <span>{errors.find((error) => error.field === 'suggestions').text}</span>}
                </section>
            </main>
        </React.Fragment>
    );
};
