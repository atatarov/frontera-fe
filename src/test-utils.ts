import {act} from "react-dom/test-utils";
import _ from 'lodash';

export const multipleRequest = (mock, responses) =>
    act(async () => {
        await mock.onAny().reply((config) => {
            if(responses.length === 0) {
                return[500, {}];
            }
            console.log({responses});
            const [method, url, params, ...response] = responses.shift()
            const match =
                config.url.includes(url) &&
                config.method.toUpperCase() === method &&
                (!config.params || _.isEqual(config.params, params))
            if (match) {
                return response
            }

            return [500, {}]
        })
    });