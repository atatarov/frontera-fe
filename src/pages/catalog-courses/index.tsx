import { CatalogCourses } from './catalog-courses'

export default CatalogCourses;