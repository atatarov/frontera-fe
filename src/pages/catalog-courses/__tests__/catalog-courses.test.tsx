import React from 'react';
import { describe, it, expect, beforeEach } from '@jest/globals';
import { mount } from 'enzyme';
import axios from 'axios';
import { Provider } from "react-redux";
import MockAdapter from "axios-mock-adapter";

import { multipleRequest } from "../../../test-utils";
import { store } from "../../../__data__/store";
import { CatalogCourses } from '../catalog-courses';

import catalogCoursesResponse from '../../../../stubs/api/mocks/suggestions/success.json';

describe('Тестируем CatalogCourses', () => {
    let mockApi;
    beforeEach(() => {
        mockApi = new MockAdapter(axios);
    });
    it('Тестируем рендер ', async () => {
        const component = mount(
            <Provider store={store}>
                <CatalogCourses />
            </Provider>
        );
        expect(component).toMatchSnapshot();
        const response = [
            ['GET', '/suggestions/success', {}, 200, { ...catalogCoursesResponse }]
        ];
        await multipleRequest(mockApi, response);
        component.update();
        expect(component).toMatchSnapshot();
    });
});