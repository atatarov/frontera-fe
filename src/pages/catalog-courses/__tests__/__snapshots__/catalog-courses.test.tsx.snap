// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем CatalogCourses Тестируем рендер  1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <CatalogCourses>
    <Header>
      <div
        className="headerWrapper"
      >
        <header
          className="header"
        >
          <nav
            className="headerNav"
          >
            <IconLogo>
              <a
                className="navLogo"
                href=""
              >
                <img
                  alt="Purple Frontera logo"
                  src="FILE_MOCK"
                />
              </a>
            </IconLogo>
            <HeaderNav>
              <Link
                as={[Function]}
                to="/home"
              >
                <Link
                  className="link"
                  to="/home"
                >
                  <a
                    className="link"
                    href="/home"
                  >
                    Главная
                  </a>
                </Link>
              </Link>
              <Link
                as={[Function]}
                to="/catalog-courses"
              >
                <Link
                  className="link"
                  to="/catalog-courses"
                >
                  <a
                    className="link"
                    href="/catalog-courses"
                  >
                    Рекомендации
                  </a>
                </Link>
              </Link>
            </HeaderNav>
          </nav>
          <AccountBar>
            <div
              className="AccountBar"
            >
              <SearchInput>
                <div
                  className="searchInput"
                >
                  <form
                    className="searchForm"
                  >
                    <label
                      htmlFor="search"
                    >
                      <input
                        id="search"
                        placeholder="Search..."
                        type="text"
                      />
                    </label>
                    <button
                      onClick={[Function]}
                      type="submit"
                    />
                  </form>
                </div>
              </SearchInput>
              <ProfileButton>
                <div
                  className="wrapper"
                >
                  <div
                    className="dropdown"
                  >
                    <button
                      className="dropdownButton"
                      onClick={[Function]}
                      type="button"
                    />
                  </div>
                </div>
              </ProfileButton>
            </div>
          </AccountBar>
        </header>
      </div>
    </Header>
    <main
      className="container"
    >
      <PageHeader
        label="На основании вашего выбора"
      >
        <h2
          className="pageHeader"
        >
          На основании вашего выбора
        </h2>
      </PageHeader>
      <section>
        <Loader>
          <div
            className="spinner"
          >
            <div
              className="bounce1"
            />
            <div
              className="bounce2"
            />
            <div
              className="bounce3"
            />
          </div>
        </Loader>
      </section>
    </main>
  </CatalogCourses>
</Provider>
`;

exports[`Тестируем CatalogCourses Тестируем рендер  2`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <CatalogCourses>
    <Header>
      <div
        className="headerWrapper"
      >
        <header
          className="header"
        >
          <nav
            className="headerNav"
          >
            <IconLogo>
              <a
                className="navLogo"
                href=""
              >
                <img
                  alt="Purple Frontera logo"
                  src="FILE_MOCK"
                />
              </a>
            </IconLogo>
            <HeaderNav>
              <Link
                as={[Function]}
                to="/home"
              >
                <Link
                  className="link"
                  to="/home"
                >
                  <a
                    className="link"
                    href="/home"
                  >
                    Главная
                  </a>
                </Link>
              </Link>
              <Link
                as={[Function]}
                to="/catalog-courses"
              >
                <Link
                  className="link"
                  to="/catalog-courses"
                >
                  <a
                    className="link"
                    href="/catalog-courses"
                  >
                    Рекомендации
                  </a>
                </Link>
              </Link>
            </HeaderNav>
          </nav>
          <AccountBar>
            <div
              className="AccountBar"
            >
              <SearchInput>
                <div
                  className="searchInput"
                >
                  <form
                    className="searchForm"
                  >
                    <label
                      htmlFor="search"
                    >
                      <input
                        id="search"
                        placeholder="Search..."
                        type="text"
                      />
                    </label>
                    <button
                      onClick={[Function]}
                      type="submit"
                    />
                  </form>
                </div>
              </SearchInput>
              <ProfileButton>
                <div
                  className="wrapper"
                >
                  <div
                    className="dropdown"
                  >
                    <button
                      className="dropdownButton"
                      onClick={[Function]}
                      type="button"
                    />
                  </div>
                </div>
              </ProfileButton>
            </div>
          </AccountBar>
        </header>
      </div>
    </Header>
    <main
      className="container"
    >
      <PageHeader
        label="На основании вашего выбора"
      >
        <h2
          className="pageHeader"
        >
          На основании вашего выбора
        </h2>
      </PageHeader>
      <section>
        <CourseCardsLarge
          cards={
            Array [
              Object {
                "id": 1,
                "imageSrc": "remote-assets/images/suggestions/sg-course-1.jpg",
                "label": "React",
              },
              Object {
                "id": 2,
                "imageSrc": "remote-assets/images/suggestions/sg-course-2.jpg",
                "label": "CI/CD",
              },
              Object {
                "id": 3,
                "imageSrc": "remote-assets/images/suggestions/sg-course-3.jpg",
                "label": "Изучаем JS",
              },
              Object {
                "id": 4,
                "imageSrc": "remote-assets/images/suggestions/sg-course-4.jpg",
                "label": "git",
              },
            ]
          }
          showNumber={false}
        >
          <ul
            className="CatalogList"
          >
            <li
              className="CardContainer"
              key="React0"
            >
              <div
                className="CatalogItemLong"
              >
                <SmartImage
                  path="remote-assets/images/suggestions/sg-course-1.jpg"
                >
                  <img
                    className="CardImage"
                    onError={[Function]}
                    src="remote-assets/images/suggestions/sg-course-1.jpg"
                  />
                </SmartImage>
                <Link
                  className="CatalogLinkLong"
                  to="/course-page/1"
                >
                  <a
                    className="CatalogLinkLong"
                    href="/course-page/1"
                  >
                    <h3
                      className="CatalogHeaderLong"
                    >
                      <span
                        className="CatalogCategoryLong"
                      >
                        React
                      </span>
                    </h3>
                  </a>
                </Link>
              </div>
            </li>
            <li
              className="CardContainer"
              key="CI/CD1"
            >
              <div
                className="CatalogItemLong"
              >
                <SmartImage
                  path="remote-assets/images/suggestions/sg-course-2.jpg"
                >
                  <img
                    className="CardImage"
                    onError={[Function]}
                    src="remote-assets/images/suggestions/sg-course-2.jpg"
                  />
                </SmartImage>
                <Link
                  className="CatalogLinkLong"
                  to="/course-page/2"
                >
                  <a
                    className="CatalogLinkLong"
                    href="/course-page/2"
                  >
                    <h3
                      className="CatalogHeaderLong"
                    >
                      <span
                        className="CatalogCategoryLong"
                      >
                        CI/CD
                      </span>
                    </h3>
                  </a>
                </Link>
              </div>
            </li>
            <li
              className="CardContainer"
              key="Изучаем JS2"
            >
              <div
                className="CatalogItemLong"
              >
                <SmartImage
                  path="remote-assets/images/suggestions/sg-course-3.jpg"
                >
                  <img
                    className="CardImage"
                    onError={[Function]}
                    src="remote-assets/images/suggestions/sg-course-3.jpg"
                  />
                </SmartImage>
                <Link
                  className="CatalogLinkLong"
                  to="/course-page/3"
                >
                  <a
                    className="CatalogLinkLong"
                    href="/course-page/3"
                  >
                    <h3
                      className="CatalogHeaderLong"
                    >
                      <span
                        className="CatalogCategoryLong"
                      >
                        Изучаем JS
                      </span>
                    </h3>
                  </a>
                </Link>
              </div>
            </li>
            <li
              className="CardContainer"
              key="git3"
            >
              <div
                className="CatalogItemLong"
              >
                <SmartImage
                  path="remote-assets/images/suggestions/sg-course-4.jpg"
                >
                  <img
                    className="CardImage"
                    onError={[Function]}
                    src="remote-assets/images/suggestions/sg-course-4.jpg"
                  />
                </SmartImage>
                <Link
                  className="CatalogLinkLong"
                  to="/course-page/4"
                >
                  <a
                    className="CatalogLinkLong"
                    href="/course-page/4"
                  >
                    <h3
                      className="CatalogHeaderLong"
                    >
                      <span
                        className="CatalogCategoryLong"
                      >
                        git
                      </span>
                    </h3>
                  </a>
                </Link>
              </div>
            </li>
          </ul>
        </CourseCardsLarge>
      </section>
    </main>
  </CatalogCourses>
</Provider>
`;
